pkg/auth/handler/signup.go

```diff
@@ -58,57 +60,49 @@ func (f SignupHandlerFactory) ProvideAuthzPolicy() authz.Policy {
 }
 
 type SignupRequestPayload struct {
-	AuthData   map[string]string      `json:"auth_data"`
-	Password   string                 `json:"password"`
-	RawProfile map[string]interface{} `json:"profile"`
+	LoginIDs map[string]string      `json:"login_ids"`
+	Password string                 `json:"password"`
+	Metadata map[string]interface{} `json:"metadata"`
 }
 
 func (p SignupRequestPayload) Validate() error {
 	if p.isAnonymous() {
 		//no validation logic for anonymous sign up
 	} else {
-		if len(p.AuthData) == 0 {
-			return skyerr.NewInvalidArgument("empty auth data", []string{"auth_data"})
-		}
-
-		if duplicatedKeys := p.duplicatedKeysInAuthDataAndProfile(); len(duplicatedKeys) > 0 {
-			return skyerr.NewInvalidArgument("duplicated keys found in auth data in profile", duplicatedKeys)
+		if len(p.LoginIDs) == 0 {
+			return skyerr.NewInvalidArgument("empty login_ids", []string{"login_ids"})
 		}
 
 		if p.Password == "" {
 			return skyerr.NewInvalidArgument("empty password", []string{"password"})
 		}
+
+		if p.duplicatedLoginIDs() {
```

Comment:
This checking for duplicated loginID, for example:
```
{
  "username": "example",
  "nickname": "example"
}
```

Reply:
Oh so it is related to https://github.com/SkygearIO/features/pull/295/files.

Eventually the uniqueness should be checked at db level, but not in request validation.

To avoid handling duplicated data later, I suggest adding the unique constraint here.

Reply:
Sorry, I do not quite understand. 

Currently, in db level, `login_id` already has `UNIQUE` constraint. So here request validation is for providing a better error message for the user to understand they input wrong `loginIDs` (duplicated values).

At db level, we will return `ErrUserDuplicated` if `loginID` duplicated to another user.

Reply:
Oh I see. I suggest add some comment here, I could not relate this validation to that feature immediately.